packages/worker/native/src/lib.rs

```diff
@@ -79,28 +83,30 @@ where
 }
 
 /// Send an error to JS via callback using an [EventQueue]
-fn send_error<T>(queue: Arc<EventQueue>, callback: Root<JsFunction>, error: T)
+fn send_error<E, F>(queue: Arc<EventQueue>, callback: Root<JsFunction>, error_ctor: F)
 where
-    T: Display + Send + 'static,
+    E: Object,
+    F: for<'a> FnOnce(&mut TaskContext<'a>) -> JsResult<'a, E> + Send + 'static,
```

Comment:
Wow, you had to use the https://doc.rust-lang.org/nomicon/hrtb.html . Rust expert already :D